### PR TITLE
adding the SCF ops required for jit64 indexing

### PR DESCRIPTION
**Pull request by kushanam**
A follow up PR to enable the mlir::scf ops and compile and execute kernels within the xla_legalize_tf_no_fallback pass. This is required to compile the jit 64 bit indexing.
cc: @frgossen

### CODE REVIEW

File: tensorflow/compiler/mlir/xla/transforms/xla_legalize_tf_no_fallback.cc [hunk 1]
```diff
@@ -63,10 +69,15 @@ void LegalizeTFNoFallback::runOnOperation() {
   target.addLegalDialect<arith::ArithmeticDialect>();
   target.addLegalDialect<chlo::ChloDialect>();
   target.addLegalDialect<MhloDialect>();
+  target.addLegalDialect<StandardOpsDialect>();
```
Comment (frgossen) [inline tensorflow/compiler/mlir/xla/transforms/xla_legalize_tf_no_fallback.cc:72]: This dialect no longer exists. You might want to rebase.

File: tensorflow/compiler/mlir/xla/transforms/xla_legalize_tf_no_fallback.cc [hunk 2]
```diff
@@ -38,6 +41,9 @@ namespace {
 
 class LegalizeTFNoFallback
     : public LegalizeTFNoFallbackBase<LegalizeTFNoFallback> {
+  void getDependentDialects(DialectRegistry &registry) const override {
+    registry.insert<scf::SCFDialect>();
```
Comment (frgossen) [inline tensorflow/compiler/mlir/xla/transforms/xla_legalize_tf_no_fallback.cc:45]: Is this needed? The `LegalizeTFNoFallback` pass will not create any scf ops, will it?

**Review by frgossen — CHANGES_REQUESTED** [review]
Thank you Kushan!

File: tensorflow/compiler/mlir/xla/transforms/xla_legalize_tf_no_fallback.cc (hunk 2 above)
Comment (kushanam) [reply tensorflow/compiler/mlir/xla/transforms/xla_legalize_tf_no_fallback.cc:45]: I think unless I added it the code doesn't allow me to legalize YieldOp and IfOp.

**Comment by kushanam** [comment]
Frederik (@frgossen) the build issue `Executing genrule //tensorflow:tf_python_api_gen_v2` seem to b non-related to current PR, unless it requires the rolled back PR to be in. Any thoughts?

**Review by frgossen — APPROVED** [review]
Thanks!

**Comment by frgossen** [comment]
> Frederik (@frgossen) the build issue `Executing genrule //tensorflow:tf_python_api_gen_v2` seem to b non-related to current PR, unless it requires the rolled back PR to be in. Any thoughts?

Can you try a rebase?

**Comment by frgossen** [comment]
Hi Kushan, I think we will have to wait until the unrelated failures are fixed. We can try rebasing and rerunning the tests.

**Pull request merged**